store/postgres/src/store.rs

```diff
@@ -665,17 +704,116 @@ impl Store {
         Ok(())
     }
 
+    /// Panics:
+    /// * This task will panic if the supplied closure panics
+    /// * This task will panic if the supplied closure returns Err(Cancelled)
+    ///   when the supplied cancel token is not cancelled.
+    // Ideally, we want to pass a borrow of the PooledConnection in. But,
+    // with_entity_conn requires this to be owned to give it to e::Connection
+    // If e::Connection could take a reference this would not be a problem
```

Comment:
What would it take to make it so that we can pass a reference of `PooledConnection` in?

Reply:
`entities::Connection`` would need to have a lifetime on it. This seems like the right thing to do. Let me do a quick dive and see how much lift it would require.